### DISCUSSION TITLE
Charger won't start TP4056

### DISCUSSION THREAD

**Discussion by kotylo** [PricelessToolkit/MailBoxGuard#35] [General]
Hi. Reaching out, maybe someone (probably PricelessToolkit) could shed a light on my issue.
I've soldered the PCB via Mini Hot Plate, which worked nicely. The reed switch also works and i get the 350-400m range on 868 antenna. What doesn't work is the charger circuit. 

When i connect the power to the Type-C connector, the red LED flashes for like 50ms and stops. No current flows.

Measurements of TP4056 pins:
|   |  TypeC ✔| Type-C ❌| 
|---|---|---|
|  Battery ✔ | BAT: 4.06<br/>Vcc: 5.08  | BAT: 4.06<br/>Vcc: 0.5V and dropping (after some time (sleep?) will be 0) |  
|  Baterry ❌ |  BAT: 2.0, and dropping.. now 1.01<br/>Vcc: 5.08 |  BAT: 0.95, and dropping<br/>Vcc: 0.57 and dropping |

I have tried to disconnect battery, but not Type-C and try to measure GND-Vcc, the Green LED did light up for a second and then faded away. Could reproduce it twice, but not anymore.

If i try to pull STDB pin low for a bit, the Green LED lights up faintly or sometimes fully, so it it not dead (yet :D)

Just tried to swap with different TP4056 chip from the same batch, same result, unfortunately.
Some photos:
![PXL_20240111_110629826](https://github.com/PricelessToolkit/MailBoxGuard/assets/2077668/26d81e5c-691d-481c-bccd-13e381eeb13b)
![PXL_20240111_110614619](https://github.com/PricelessToolkit/MailBoxGuard/assets/2077668/cbf79160-8aa7-4c69-8dd7-c3b79b7e64eb)
![PXL_20240111_110658963](https://github.com/PricelessToolkit/MailBoxGuard/assets/2077668/8c40b6f5-ba88-4ba9-b827-1b2d29316fe2)

**Comment by PricelessToolkit** [comment]
1. "When I connect the power to the Type-C connector, the red LED flashes for like 50ms and stops. No current flows." - It's OK if no battery
2. You can't power the MailBox sensor only with a USB
3.  VCC for you is which pad?

I can't see any anomaly in this

|   |  TypeC ✔| Type-C ❌| 
|---|---|---|
|  Battery ✔ | BAT: 4.06<br/>Vcc: 5.08  | BAT: 4.06<br/>Vcc: 0.5V and dropping (after some time (sleep?) will be 0) |  
|  Baterry ❌ |  BAT: 2.0, and dropping.. now 1.01<br/>Vcc: 5.08 |  BAT: 0.95, and dropping<br/>Vcc: 0.57 and dropping |


It's not charging ??

**Reply by kotylo** [reply]
1. With battery or without, same thing: red LED flashes barely and stops
2. Yes, of course. The goal is to charge the battery :) I went around the block with just the battery, it's impressive: 868Mhz antenna with 350m-400m range for me, through 2-3 buildings or so.
3. Vcc of the TP4056, so the pad `#4` (edited, thanks)

Yes, not charging, so 4.05 on battery terminals. I have also one more 100% working TP4056 module with protection, i can try to desolder it tomorrow, in case there is no visible ideas to try otherwise. And thank you so much for helping! Have a good night ;)

![PXL_20240112_020010645](https://github.com/PricelessToolkit/MailBoxGuard/assets/2077668/f86a49db-3c5b-4ea4-aa7b-56622d45c6b1)

**Reply by PricelessToolkit** [reply]
can you discharge the battery to 3.8-3.9 and test again? TP4056 probably thinks it's already fully charged.

**Reply by PricelessToolkit** [reply]
Vcc of the TP4056, so the pad https://github.com/PricelessToolkit/MailBoxGuard/issues/4 :) 🥇

**Reply by kotylo** [reply]
Discharged to 3.9V, same thing :( Usually this chip will light up at least Green LED constantly if it's fully charged, so far i remember from other PCBs.

![image](https://github.com/PricelessToolkit/MailBoxGuard/assets/2077668/fea79e80-325a-4d89-b10c-8634e24d2d93)

So i don't get this chip)))
- Vin 5V = ✔,
- TEMP pin is Grounded = temp sensing is disabled ✔
- No battery = there is a battery ✔

**Reply by PricelessToolkit** [reply]
It's strange! probably TP4056 is a dead "bad batch?" Maybe the last thing is "Clean pcb? there may be a short circuit under the resistors " I can't say anything else :( sorry, If you find the reason please let me know, I love such mystical problems.

**Reply by kotylo** [reply]
yeah, it calms me down that you can't find it) nice idea with short. I'll try those things and get back to you:
- transferring the working chip
- desolder resistors and check for shorts underneath (there was a bit more soldering paste that i wanted when i started, so it's a possibility)
- i have 4 more PCBs (of course), so i'll try them out too. But wanted to make sure i don't do the same mistake twice beforehand

**Reply by kotylo** [reply]
![find-2](https://github.com/PricelessToolkit/MailBoxGuard/assets/2077668/4b6a540a-6a69-4cfe-a562-3572426aa785)
Find 2 differences...  Thanks for idea to check for the short circtuit, lol :)
<details><summary>🤦🏻‍♂️</summary>
<p>
So i went with checking for resistor's values at first and one was "shorted". But when i desoldered it, surprise-surprise - there lies a path underneath by design! Could PCBWay have made a mistake? How is that even possible? Wh... did <b>I</b> do?! Omg... In spite of your nice BOM file, schematics, in spite of this beautiful design with white lines everywhere, i ef upd.
</p>
</details>

**Reply by PricelessToolkit** [reply]
It happens... https://github.com/PricelessToolkit/MailBoxGuard/issues/4 :) the first photo in this discussion "problem is visible very well" but I didn't notice either. the main thing is that the problem was found!
